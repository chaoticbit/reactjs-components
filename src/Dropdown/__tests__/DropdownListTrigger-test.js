/* eslint-disable no-unused-vars */
import React from "react";
/* eslint-enable no-unused-vars */
import renderer from "react-test-renderer";
import DropdownListTrigger from "../DropdownListTrigger.js";

var TestUtils;
if (React.version.match(/15.[0-5]/)) {
  TestUtils = require("react-addons-test-utils");
} else {
  TestUtils = require("react-dom/test-utils");
}

describe("DropdownListTrigger", function() {
  it("renders correctly with given props", function() {
    const tree = renderer
      .create(
        <DropdownListTrigger
          onTrigger={this.callback}
          placeholder="Placeholder"
          disabled
          className="class-1 class-2"
          selectedItem={{
            html: "html",
            id: "id",
            selectedHtml: "selectedHtml"
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders corectly with null item", function() {
    const tree = renderer
      .create(
        <DropdownListTrigger
          onTrigger={this.callback}
          placeholder="Placeholder"
          disabled
          className="class-1 class-2"
          selectedItem={null}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with empty item", function() {
    const tree = renderer
      .create(
        <DropdownListTrigger
          onTrigger={this.callback}
          placeholder="Placeholder"
          disabled
          className="class-1 class-2"
          selectedItem={{
            id: "id"
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly without props", function() {
    const tree = renderer.create(<DropdownListTrigger />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe("#onTrigger", function() {
    afterEach(function() {
      Array.prototype.slice
        .call(document.body.children)
        .forEach(function(node) {
          document.body.removeChild(node);
        });
    });
    it("triggers onTrigger", function() {
      const callback = jest.fn();
      const instance = TestUtils.renderIntoDocument(
        <DropdownListTrigger onTrigger={callback} />
      );
      const button = TestUtils.findRenderedDOMComponentWithTag(
        instance,
        "button"
      );
      TestUtils.Simulate.click(button);
      expect(callback).toBeCalled();
    });
    it("triggers onTrigger", function() {
      const callback = jest.fn();
      const instance = TestUtils.renderIntoDocument(
        <DropdownListTrigger disabled onTrigger={callback} />
      );
      const button = TestUtils.findRenderedDOMComponentWithTag(
        instance,
        "button"
      );
      TestUtils.Simulate.click(button);
      expect(callback).not.toBeCalled();
    });
  });
});
