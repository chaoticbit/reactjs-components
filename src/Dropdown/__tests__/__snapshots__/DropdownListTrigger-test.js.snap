exports[`DropdownListTrigger renders corectly with null item 1`] = `
<button
  className="class-1 class-2"
  disabled={true}
  onClick={[Function]}
  type="button">
  Placeholder
</button>
`;

exports[`DropdownListTrigger renders correctly with empty item 1`] = `
<button
  className="class-1 class-2"
  disabled={true}
  onClick={[Function]}
  type="button" />
`;

exports[`DropdownListTrigger renders correctly with given props 1`] = `
<button
  className="class-1 class-2"
  disabled={true}
  onClick={[Function]}
  type="button">
  selectedHtml
</button>
`;

exports[`DropdownListTrigger renders correctly without props 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
  type="button" />
`;
